tests/integration/operatorhub/cmd_service_test.go

```diff
@@ -68,33 +68,33 @@ var _ = Describe("odo service command tests for OperatorHub", func() {
 
 			When("a nodejs component is created", func() {
 
-				JustBeforeEach(func() {
+				BeforeEach(func() {
 					helper.Cmd("odo", "create", "nodejs").ShouldPass().Out()
 				})
 
-				JustAfterEach(func() {
+				AfterEach(func() {
 					// we do this because for these specific tests we dont delete the project
 					helper.Cmd("odo", "delete", "--all", "-f").ShouldPass().Out()
 				})
 
 				When("creating a postgres operand with params", func() {
 					var operandName string
 
-					JustBeforeEach(func() {
+					BeforeEach(func() {
 						operandName = helper.RandString(10)
 						helper.Cmd("odo", "service", "create", postgresDatabase, operandName, "-p",
 							"databaseName=odo", "-p", "size=1", "-p", "databaseUser=odo", "-p",
 							"databaseStorageRequest=1Gi", "-p", "databasePassword=odopasswd").ShouldPass().Out()
 
 					})
 
-					JustAfterEach(func() {
+					AfterEach(func() {
 						helper.Cmd("odo", "service", "delete", fmt.Sprintf("Database/%s", operandName), "-f").ShouldPass().Out()
 						helper.Cmd("odo", "push").ShouldPass().Out()
 					})
 
 					When("odo push is executed", func() {
-						JustBeforeEach(func() {
+						BeforeEach(func() {
```

Comment:
@feloy I see multiple nested `when` (from L55-L115) but they are testing only two `IT` spec which are at the same level, are these in this way to avoid extra work to add new test case if required in future?

Reply:
Girish have written this part, but I think it is interesting to keep the different steps as nested Whens, so we can see more clearly what are the different steps executed by the user, and also it will be easier to add more tests between steps in the future.

Generally, I put an odo command and the following `odo push` in the same When if I can't see any `It` that could be added between the odo command and `odo push`.

Reply:
understood, and make sense! 
Thanks @feloy